src/gdb.rs

```diff
@@ -1,13 +1,55 @@
-use gdbstub::arch::{Arch, SingleStepGdbBehavior};
+use core::num::NonZeroUsize;
+use gdbstub::arch::{Arch, RegId, Registers};
 use gdbstub::common::Signal;
 use gdbstub::target;
 use gdbstub::target::ext::base::singlethread::{SingleThreadBase, SingleThreadResume};
 use gdbstub::target::{Target, TargetResult};
 
 use crate::emu::{Emu, ExecMode};
 
+#[derive(Debug)]
+pub struct GenericRegId(#[allow(dead_code)] pub usize);
+
+impl RegId for GenericRegId {
+    fn from_raw_id(id: usize) -> Option<(Self, Option<NonZeroUsize>)> {
+        Some((GenericRegId(id), None))
+    }
+}
+
+#[derive(Debug, Default, Clone, PartialEq)]
+pub struct GenericRegs {
+    pub dummy: u64,
```

Comment:
are you sure that this totally generic register representation will suffice to accurately hit the codepaths we're interested in validating? I worry that other code in GDB might get confused when it sees all-zero responses to these various registers.

Then again - its not like the previous code did all that much work to make these registers appear "reasonable" either... so idk

Reply:
I'm not 100% certain. Anyway, this code produces exactly the same behavior as the previous code, and the new cases also match the behavior I tested in the actual environment. I think this code is sufficient for getting upstream's attention.

Reply:
Sounds good. Happy to merge as is, and iterate more in the future, if need be.